R/insertPackage.R

```diff
@@ -159,75 +159,68 @@ identifyPackageType <- function(file) {
     return(ret)
 }
 
+##' This function returns the compile-time information added
+##' to the \code{DESCRIPTION} file in the package.
+##'
+##' @title Get information from a binary package 
+##' @param file the fully qualified path of the package
+##' @return A named vector with several components
+##' @author Dirk Eddelbuettel
+getPackageInfo <- function(file) {
+    if (!file.exists(file)) stop("File ", file, " not found!", call.=FALSE)
+    
+    td <- tempdir()
+    if (grepl(".zip$", file)) {
+        unzip(file, exdir=td)
+    } else if (grepl(".tgz$", file)) {
+        untar(file, exdir=td)
+    } else {
+        ##stop("Not sure we can handle ", file, call.=FALSE)
+        fields <- c("Source"=TRUE, "Rmajor"=NA, "Mavericks"=FALSE)
+        return(fields)
+    }
+
+    pkgname <- gsub("^([a-zA-Z0-9]*)_.*", "\\1", basename(file))
+    path <- file.path(td, pkgname, "DESCRIPTION")
+    builtstring <- read.dcf(path, 'Built')
+    unlink(file.path(td, pkgname), recursive=TRUE)
+
+    fields <- strsplit(builtstring, "; ")[[1]]
+    names(fields) <- c("Rversion", "OSflavour", "Date", "OS")
+
+    rmajor <- gsub("^R (\\d\\.\\d)\\.\\d.*", "\\1", fields["Rversion"])
+    isDarwin13 <- ifelse(fields["OSflavour"] == "x86_64-apple-darwin13.4.0", "yes", "no")
+    fields <- c(fields, "Rmajor"=unname(rmajor), "Mavericks"=unname(isDarwin13))
```

Comment:
This is indeed the crux of the matter.  We could just invert and test for 'non-Maverick' by honing in on the older symbol (ie 10.8.0) or empty string for non-binary packages:

```sh
$ ./checkPackages.r
digest_0.6.8.r-oldrel.snowleopard.tgz --> x86_64-apple-darwin10.8.0 
digest_0.6.8.r-release.mavericks.tgz --> x86_64-apple-darwin13.4.0 
digest_0.6.8.r-release.snowleopard.tgz --> x86_64-apple-darwin10.8.0 
drat_0.0.3.r-oldrel.snowleopard.tgz -->  
drat_0.0.4.r-release.mavericks.tgz -->  
drat_0.0.4.r-release.snowleopard.tgz -->  
$ 
```

Thoughts?